source/h1_stream.c

```diff
@@ -131,6 +126,10 @@ struct aws_h1_stream *aws_h1_stream_new_request_handler(const struct aws_http_re
         return NULL;
     }
 
+    /* big assumption, in server mode, the following code can only run from the event-loop thread so don't worry
```

Comment:
Should we assert on this? Feels like we should?

Reply:
what would I assert on?

Reply:
the other good news for Http 1.1. nothing bad can happen on a 32 bit integer that's only incrementing.

Reply:
Actually, it's not an assumption at all.
This function **can** only be called on the event-loop thread 🥳
The function that calls this one h1_connection.c raises an error if that's not the case.

You could assert again here to drive home the point, or modify the comment to state this as fact